cni/pkg/install-cni/cmd/root.go

```diff
@@ -0,0 +1,131 @@
+	registerStringParameter(constants.CNIConfName, "", "Name of the CNI configuration file")
+	registerBooleanParameter(constants.ChainedCNIPlugin, true, "Whether to install CNI plugin as a chained or standalone")
+	registerStringParameter(constants.CNINetworkConfig, "", "CNI config template as a string")
+
+	// Not configurable in CNI helm charts
+	registerStringParameter(constants.MountedCNINetDir, "/host/etc/cni/net.d", "Directory on the container where CNI networks are installed")
+	registerBooleanParameter(constants.Sleep, true,
+		"Whether to sleep and wait for modifications after installation to prevent Kubernetes from restarting the pod repeatedly")
+	registerStringParameter(constants.CNINetworkConfigFile, "", "CNI config template as a file")
+	registerStringParameter(constants.LogLevel, "warn", "Fallback value for log level in CNI config file, if not specified in helm template")
+	registerStringParameter(constants.KubecfgFilename, "ZZZ-istio-cni-kubeconfig", "Name of the kubeconfig file")
+	registerIntegerParameter(constants.KubeconfigMode, constants.DefaultKubeconfigMode, "File mode of the kubeconfig file")
+	registerStringParameter(constants.KubeCAFile, "", "CA file for kubeconfig. Defaults to the pod one")
+	registerBooleanParameter(constants.SkipTLSVerify, false, "Whether to use insecure TLS in kubeconfig file")
+	registerBooleanParameter(constants.UpdateCNIBinaries, true, "Update binaries")
+	registerStringArrayParameter(constants.SkipCNIBinaries, []string{}, "Binaries that should not be installed")
+}
+
+func registerStringParameter(name, value, usage string) {
+	rootCmd.Flags().String(name, value, usage)
+	bindViper(name)
+}
+
+func registerStringArrayParameter(name string, value []string, usage string) {
+	rootCmd.Flags().StringArray(name, value, usage)
+	bindViper(name)
+}
+
+func registerIntegerParameter(name string, value int, usage string) {
+	rootCmd.Flags().Int(name, value, usage)
+	bindViper(name)
+}
+
+func registerBooleanParameter(name string, value bool, usage string) {
+	rootCmd.Flags().Bool(name, value, usage)
+	bindViper(name)
+}
+
+func bindViper(name string) {
+	if err := viper.BindPFlag(name, rootCmd.Flags().Lookup(name)); err != nil {
+		log.Errora(err)
+		os.Exit(1)
+	}
+}
+
+func constructConfig() (*config.Config, error) {
+	cfg := &config.Config{
+		CNINetDir:        viper.GetString(constants.CNINetDir),
+		MountedCNINetDir: viper.GetString(constants.MountedCNINetDir),
+		CNIConfName:      viper.GetString(constants.CNIConfName),
+		ChainedCNIPlugin: viper.GetBool(constants.ChainedCNIPlugin),
+		Sleep:            viper.GetBool(constants.Sleep),
+
+		CNINetworkConfigFile: viper.GetString(constants.CNINetworkConfigFile),
+		CNINetworkConfig:     viper.GetString(constants.CNINetworkConfig),
+
+		LogLevel:           viper.GetString(constants.LogLevel),
+		KubeconfigFilename: viper.GetString(constants.KubecfgFilename),
```

Comment:
There is some inconsistency for naming Kube Config Filename.
`KubeconfigFilename`
`KubecfgFilename`
 
It will be good to name it `KubeConfigFilename` or `KubeCfgFilename` ?